Connection error logging in PersistentConnection.cs not surfacing inner exception message

gobriango19 opened the pull request:
When PersistentConnection.cs logs connection failures, it only logs the outer exception message. The outer exception often obfuscates the actual connection error captured by the inner exception. For connection errors debugging we are finding the outer exception much less helpful than inner exception, and I think we should surface the inner exception message when an inner exception is present.

zidad commented:
I like it, but should we maybe concatenate the messages not to lose the message of the outer exception one? Do you have an example of the values that were wrong?

Could you also bump the version in version.cs and add a small description? 

Thanks a lot!

gobriango19 commented:
I have updated PersistentConnection.cs so that both outer and inner exception messages are logged in the following format: _outer exception (inner exception)_. I have also updated the Version.cs file.

The inner exception message is often useful when debugging TLS connections. For example, in the following three instances both outer exceptions have the same message, but their inner exception messages are different:
- None of the specified endpoints were reachable (connection.start was never received, likely due to a network timeout)
- None of the specified endpoints were reachable (Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host.)
- None of the specified endpoints were reachable (A call to SSPI failed, see inner exception.)

Here is also a blog post documenting the author's usage of RabbitMQ client's inner exception to debug connection issues: http://thoughtsoncoding.blogspot.co.nz/2014/03/rabbitmq-net-and-ssl-easier-said-than.html

Cheers!

zidad commented:
@gobriango19 I like it, thanks a lot for your contribution!
